File: my-folder/1341-split-a-string-in-balanced-strings/solution.java
class Solution {
    public int balancedStringSplit(String s) {
        int count = 0;
        int l = 0;
        int r = 0;
    
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == 'L'){
                l++;
            } else {
                r++;
            }
            if(l != 0 && l == r){
                count++;
                l = 0;
                r = 0;
            }
        }
        return count;
    }
}

File: my-folder/1341-split-a-string-in-balanced-strings/solution.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        ans = 0
        l = 0
        r = 0
        
        for c in s:
            if c == 'R':
                r += 1
            else:
                l += 1
            
            if l == r:
                ans += 1
                
        return ans
